homeassistant/components/homeassistant_hardware/silabs_multiprotocol_addon.py

```diff
@@ -306,6 +323,71 @@ async def _async_install_addon(self) -> None:
                 self.flow_manager.async_configure(flow_id=self.flow_id)
             )
 
+    async def _async_uninstall_addon(
+        self, addon_manager: AddonManager, *, continue_flow: bool = True
+    ) -> None:
+        """Uninstall an add-on."""
+        try:
+            try:
+                info = await addon_manager.async_get_addon_info()
+            except AddonError:
+                info = None
+
+            # Do not try to uninstall an addon if it is already uninstalled
+            if info is not None and info.state == AddonState.NOT_INSTALLED:
+                return
+
+            await addon_manager.async_uninstall_addon()
+            await self._async_wait_until_addon_state(
+                addon_manager, AddonState.NOT_INSTALLED
+            )
+        finally:
+            if continue_flow:
+                # Continue the flow after show progress when the task is done.
+                self.hass.async_create_task(
+                    self.flow_manager.async_configure(flow_id=self.flow_id)
+                )
+
+    async def _async_start_addon(
+        self, addon_manager: AddonManager, wait_until_done: bool = False
+    ) -> None:
+        """Start an add-on."""
+        try:
+            await addon_manager.async_schedule_start_addon()
+            await self._async_wait_until_addon_state(addon_manager, AddonState.RUNNING)
+
+            if wait_until_done:
+                await self._async_wait_until_addon_state(
+                    addon_manager, AddonState.NOT_RUNNING
+                )
```

Comment:
This is a confusing way to say we want to wait for an add-on first start, and then kill itself.
I think it would be clearer to explicitly first wait for state running, and then wait for state not_running when flashing.